test/CMakeLists.txt

```diff
@@ -361,247 +361,253 @@ unset(_sde_find_name)
-tbb_add_test(SUBDIR tbb NAME test_continue_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_eh_flow_graph DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_flow_graph DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_flow_graph_priorities DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_flow_graph_whitebox DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_indexer_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_join_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_join_node_key_matching DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_join_node_msg_key_matching DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_limiter_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_priority_queue_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_queue_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_sequencer_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_split_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_tagged_msg DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_overwrite_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_write_once_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_async_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_input_node DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_profiling DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_concurrent_queue_whitebox DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_intrusive_list DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_semaphore DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_environment_whitebox DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_hw_concurrency DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_eh_thread DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_global_control DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_task DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_concurrent_monitor DEPENDENCIES TBB::tbb)
-tbb_add_test(SUBDIR tbb NAME test_scheduler_mix DEPENDENCIES TBB::tbb)
-
-# test_handle_perror
-tbb_add_test(SUBDIR tbb NAME test_handle_perror)
-target_include_directories(test_handle_perror PRIVATE
-    $<TARGET_PROPERTY:TBB::tbb,INTERFACE_INCLUDE_DIRECTORIES>
-)
-
-# HWLOC related test
-tbb_add_tbbbind_test(SUBDIR tbb NAME test_arena_constraints)
-
-if (NOT "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "mips")
-    # TODO: Fix for MIPS
-    tbb_add_test(SUBDIR tbb NAME test_tbb_fork DEPENDENCIES TBB::tbb)
-endif()
-
-tbb_add_test(SUBDIR tbb NAME test_tbb_header DEPENDENCIES TBB::tbb)
-target_sources(test_tbb_header PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/tbb/test_tbb_header_secondary.cpp)
-if (TBB_OPENMP_FLAG AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "(mips)")
-    tbb_add_test(SUBDIR tbb NAME test_openmp DEPENDENCIES TBB::tbb)
+if (TBB_BUILD)
```

Comment:
I have some doubts regarding using variable with such name here. They are pretty nice in scope of including/excluding `add_subdirectory`, but considering `test/CMakeLists.txt` such names don't describe the behavior precisely for cases of testing _external_ TBB. It may be more precisely to use something like `if (TARGET TBB::tbb)` here, what do you think?

Reply:
Seems reasonable